src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -124,21 +121,23 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.AddSingleton<IHostedService>(s => s.GetRequiredService<WebJobsScriptHostService>());
 
             // Configuration
-            services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<ScriptApplicationHostOptions>, ScriptApplicationHostOptionsSetup>());
+            services.ConfigureOptions<ScriptApplicationHostOptionsSetup>();
+            services.ConfigureOptions<StandbyOptionsSetup>();
             services.ConfigureOptions<LanguageWorkerOptionsSetup>();
 
             services.TryAddSingleton<IDependencyValidator, DependencyValidator>();
         }
 
         private static void AddStandbyServices(this IServiceCollection services)
         {
-            services.AddSingleton<IOptionsChangeTokenSource<ScriptApplicationHostOptions>, StandbyChangeTokenSource>();
+            //services.AddSingleton<IOptionsChangeTokenSource<ScriptApplicationHostOptions>, StandbyChangeTokenSource<ScriptApplicationHostOptions>>();
+            services.AddSingleton<IOptionsChangeTokenSource<StandbyOptions>, StandbyChangeTokenSource>();
 
             // Core script host service
             services.AddSingleton<IHostedService>(p =>
             {
-                var hostEnvironment = p.GetService<IScriptWebHostEnvironment>();
-                if (hostEnvironment.InStandbyMode)
+                var standbyOptions = p.GetService<IOptionsMonitor<StandbyOptions>>();
+                if (standbyOptions.CurrentValue.InStandbyMode)
```

Comment:
If the options are initialized as InStandbyMode=False and earlier in the pipeline we initialize temp placeholder paths, and before we get here we are specialized (and the options are reset), doesn't the same bug exist? What prevents that?

Reply:
Ah, I remember now. The only way that this value can be reset is if the StandbyManager resets it -- and that can only happen if the host has already started (it's in an `IHostedService`). So we know it won't change during initialization, unlike the environment variables, which can change at any time.

Reply:
Ok. Recommend adding those details to the summary comments I asked for above :)

Reply:
Agreed.

Reply:
👍